### ISSUE TITLE
[BUG] Cold-compiled TVM-FFI kernels call torch.cuda.is_available() on every invocation

### ISSUE THREAD

**Issue by LeiWang1999** [tile-ai/tilelang#2684]
### Required prerequisites

- [x] I have read the documentation at https://tilelang.com.
- [x] I searched the issue tracker and did not find an existing report.

### Version and environment

- TileLang: **0.1.12+cuda.gite2db9ba5**
- The same code exists on current main at **4f442a395ea7bd16c9093e17bd029c9a93f20c59**
- Source installation, Linux AArch64, Python 3.12.11
- PyTorch 2.10.0+cu130, torch-npu 2.10.0+git6893d1e
- Ascend CANN 9.1 development build
- TVM-FFI execution backend

### Problem

A TVM-FFI kernel produced by an in-process cache miss has a large, persistent host dispatch overhead on Ascend. Warmup does not remove it.

The cause is the executable lookup in [TVMFFIKernelAdapter._convert_torch_func()](https://github.com/tile-ai/tilelang/blob/4f442a395ea7bd16c9093e17bd029c9a93f20c59/tilelang/jit/adapter/tvm_ffi.py#L199-L216):

~~~python
def get_executable():
    if self.executable is not None:
        return self.executable

    device_key: int | str = "cpu"
    if torch.cuda.is_available():
        device_key = torch.cuda.current_device()

    executable = self._executables_by_device.get(device_key)
    if executable is not None:
        return executable

    with self._executable_lock:
        executable = self._executables_by_device.get(device_key)
        if executable is None:
            executable = self._make_executable()
            self._executables_by_device[device_key] = executable
        return executable
~~~

On the cold-compilation path, self.executable starts as None. The first invocation stores the new executable only in _executables_by_device, while self.executable remains None. Every later invocation therefore calls torch.cuda.is_available() before hitting _executables_by_device.

On this Ascend environment, torch.cuda.is_available() costs approximately **80-95 us per call**, so it dominates short kernels. Warmup cannot remove a check executed on every dispatch.

The disk-cache load path behaves differently: from_database() assigns the loaded module directly to adapter.executable. It returns at the first check and does not probe CUDA. The same kernel is therefore slow in the process that compiled it and fast after restarting and loading it from disk.

Measured for the same short Ascend GEMM:

~~~text
First process, in-process cold compilation:
  public JITKernel call:       123-198 us
  underlying executable call: 13.7-14.2 us

Second process, disk cache hit:
  public JITKernel call:       17.2 us

Native reference call:         approximately 20 us
torch.cuda.is_available():     79.5 us/call over 1000 calls
~~~

For long kernels the host probe is partially hidden by asynchronous device execution, making the regression appear shape-dependent.

### Reproduction

This occurs with any short Ascend kernel compiled with execution_backend="tvm_ffi". After constructing a newly compiled kernel and NPU inputs:

~~~python
import torch

kernel(*inputs)
torch.npu.synchronize()

adapter = kernel.adapter
assert adapter.executable is None
assert adapter._executables_by_device

original = torch.cuda.is_available
cuda_probe_count = 0

def counted_is_available():
    global cuda_probe_count
    cuda_probe_count += 1
    return original()

torch.cuda.is_available = counted_is_available
try:
    for _ in range(100):
        kernel(*inputs)
    torch.npu.synchronize()
finally:
    torch.cuda.is_available = original

print(cuda_probe_count)  # 100; expected 0 on an Ascend launch path
~~~

Restarting and constructing the same kernel from TileLang's disk cache makes adapter.executable non-None and the probe count becomes zero.

### Expected behavior

Warm in-process calls after cold compilation should reuse the executable without probing CUDA. Cold-compiled and disk-cache-loaded adapters should have equivalent steady-state lookup overhead. An Ascend launch path should not invoke CUDA availability APIs.

### Proposed fix

Use one lazily initialized self.executable for both the cold-compilation and disk-cache paths, and remove the redundant Python per-device cache:

~~~python
def get_executable():
    executable = self.executable
    if executable is not None:
        return executable

    with self._executable_lock:
        executable = self.executable
        if executable is None:
            executable = self._make_executable()
            self.executable = executable
        return executable
~~~

Then _executables_by_device and the per-call torch.cuda.is_available() / torch.cuda.current_device() checks can be removed. The second check under the lock prevents duplicate initialization by concurrent callers.

This is consistent with the disk-cache path, which already stores one globally loaded runtime module in adapter.executable. It is also consistent with TVM runtime device handling: CUDAModuleNode maintains its own per-device module_[device_id] and function caches and resolves the current device during launch. The Python runtime.Executable is a JIT/link wrapper around that runtime module, not the owner of one CUDA device context.

Relevant runtime implementation:

- src/runtime/cuda/cuda_module.cc: CUDAModuleNode::GetFunc(device_id, ...) loads and caches modules per device.
- The CUDA launch function queries cudaGetDevice() and indexes its per-device function cache.

Suggested regression tests:

1. Repeated launches through a cold-compiled adapter must not call torch.cuda.is_available().
2. Cold-compiled and disk-cache-loaded adapters should both have non-None self.executable and equivalent lookup behavior.
3. One executable should be exercised across two CUDA devices to verify the runtime's existing internal per-device module/function caches.
